tensorflow/lite/micro/kernels/reduce_test.cc

```diff
@@ -126,6 +126,48 @@ void TestMeanFloatInput4D(const int* input_dims_data, const float* input_data,
                             output_data, output_dims_count, params, tolerance));
 }
 
+template <typename T>
+void TestMeanOpQuantized(const int* input_dims_data, const float* input_data,
+                         float input_scale, int input_zero_point,
+                         const int* axis_dims_data, const int32_t* axis_data,
+                         const int* output_dims_data,
+                         const float* expected_output_data, float output_scale,
+                         int output_zero_point, TfLiteReducerParams* params) {
+  // Convert dimesion arguments to TfLiteArrays
+  TfLiteIntArray* input_dims = IntArrayFromInts(input_dims_data);
+  TfLiteIntArray* axis_dims = IntArrayFromInts(axis_dims_data);
+  TfLiteIntArray* output_dims = IntArrayFromInts(output_dims_data);
+
+  // Get number of elements in input and output tensors
+  const int output_dims_count = ElementCount(*output_dims);
+  const int input_dims_count = ElementCount(*input_dims);
+
+  // Allocate arrays for quantized tensors
+  T* output_data_quant = new T[output_dims_count];
```

Comment:
Looks like these dynamic allocation has problem in some targets.
If I change this to "T output_data_quant[100];", the error disappears.
However, fixing the array length to some numeric value might not be a correct way to handle this.
@advaitjain Do you know why this happens? And is there any proper way to allocate dynamically in tf micro targets?

Reply:
Any update on this? @hajuho @advaitjain

Reply:
@petewarden Can you give a guide for allocating temporary buffers like this?

Reply:
Sorry for the delay. I'd actually recommend you pass in the output data array as an argument into the function, since the calling code will know the size needed as a constant. It's a bit less convenient, but not all of our platforms support memory allocation, so we can't use new/malloc/etc.